solidity/contracts/bridge/WalletProposalValidator.sol

```diff
@@ -645,10 +648,37 @@ contract WalletProposalValidator {
             "Target wallets do not match target wallets commitment hash"
         );
 
-        // Make sure the proposed fee is valid.
-        (uint64 movingFundsTxMaxTotalFee, , , , , , , , , , ) = bridge
-            .movingFundsParameters();
+        (
+            uint64 movingFundsTxMaxTotalFee,
+            uint64 movingFundsDustThreshold,
+            ,
+            ,
+            ,
+            ,
+            ,
+            ,
+            ,
+            ,
+
+        ) = bridge.movingFundsParameters();
+
+        // Make sure the source wallet balance is correct.
+        uint64 sourceWalletBtcBalance = getWalletBtcBalance(
+            sourceWallet.mainUtxoHash,
+            walletMainUtxo
+        );
 
+        require(
+            sourceWalletBtcBalance > 0,
+            "Source wallet BTC balance is zero"
+        );
```

Comment:
Isn't this check redundant? The `Bridge` enforces that `movingFundsDustThreshold > 0` so if the next check is fulfilled, that implies the balance is greater than zero.

Reply:
True. Removed the check in 8f8de021a29e4e408e7933783480ac8f2d8d93d9.